### PR TITLE
Define the repository scope

### PR DESCRIPTION
**Pull request by trask**
This is an attempt to define the scope of this repository, mostly in relation to https://github.com/open-telemetry/opentelemetry-java, in order to help things to go to the right place.

If this scope is agreed upon, we can move https://github.com/open-telemetry/opentelemetry-java/tree/master/extensions/runtime_metrics over to this repository, and delete https://github.com/open-telemetry/opentelemetry-java/tree/master/extensions/logging/log4j2_extensions which was accidentally re-implemented in this repository (#735).

cc: @open-telemetry/java-approvers

Will wait for @jkwatson (back Jul 30) before making any decisions here, just want to get the ball rolling and hear any initial comments / feedback.

### CODE REVIEW

File: docs/scope.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Scope of this repository
+
+Both javaagent and library-based approaches to the following:
+
+* Instrumentation for specific Java libraries and frameworks
+  * Emitting spans and metrics (and in the future logs)
+* System metrics 
+* Logging integrations
+* Spring integrations
```
Comment (anuraaga) [inline docs/scope.md:9]:
```suggestion
* Framework integrations (e.g., Spring)
```

Honestly nothing comes to mind but there must be more frameworks than just Spring right? :D

Comment (iNikem) [reply docs/scope.md:9]: E.g. Ratpack, Play :)

Comment (trask) [reply docs/scope.md]: I think frameworks are covered under Instrumentation bullet above. Narrowed this down to Spring Boot starters for now(?)

**Comment by trask** [comment]
One thing I didn't include, because I think may make more sense in SDK repo:

* Integrations with other telemetry libraries, e.g.
  * OpenTracing
  * Micrometer
  * Spring Cloud Sleuth / Brave

but we can discuss also

**Comment by trask** [comment]
Discussed with opentelemetry-java maintainers on Fri, and have agreement on this initial scope

**Pull request merged**